specreduce/wavelength_calibration.py

```diff
@@ -115,15 +129,20 @@ def __init__(self, input_spectrum, matched_line_list=None, line_pixels=None,
             if not isinstance(line_wavelengths, (u.Quantity, QTable)):
                 raise ValueError("line_wavelengths must be specified as an astropy.units.Quantity"
                                  " array or as an astropy.table.QTable")
+
+            # make sure wavelengths (or freq) are monotonic and add wavelengths
+            # to _matched_line_list
             if isinstance(line_wavelengths, u.Quantity):
-                # Ensure frequency is descending or wavelength is ascending
-                if str(line_wavelengths.unit.physical_type) == "frequency":
-                    line_wavelengths[::-1].sort()
-                else:
-                    line_wavelengths.sort()
+                if not _check_arr_monotonic(line_wavelengths):
+                    if str(line_wavelengths.unit.physical_type) == "frequency":
+                        raise ValueError('Frequencies must be strictly increasing or decreasing.')
+                    raise ValueError('Wavelengths must be strictly increasing or decreasing.')
```

Comment:
Not for this PR, but I wonder if we should rename `line_wavlengths` to `line_spectral_values` or something?

Reply:
yeah, also we should not be checking strictly for a 'wavelength' column if frequencies are allowed.